### ISSUE TITLE
Troubles with API

### ISSUE THREAD

**Issue by psi-gh** [onmyway133/DeepDiff#46]
For example, I have  and 

> protocol BaseClassProtocol { }
> class ConcreteClass1: BaseClassProtocol { } 
> class ConcreteClass2: BaseClassProtocol { }

In my code, there is only `BaseClassProtocol` using. For example:

> let array:[BaseClassProtocol]

this array can contain various classes/structs with the same protocol (obviously).
How can i apply DeepDiff library and DiffAware protocol to compare various arrays of `BaseClassProtocol` objects?

**Comment by Voxar** [comment]
Create a wrapper struct that implements DiffAware

```swift
struct DiffWrapper: DiffAware {
  var item: BaseClassProtocol
  var diffId: String {
    return item.id
  }
  static func compareContent(_ a: DiffWrapper, _ b: DiffWrapper) -> Bool {
    return a.item == b.item
  }
}
let changes = diff(old.map(DiffWrapper.init), new.map(DiffWrapper.init))
```

**Comment by psi-gh** [comment]
@Voxar Yeah, I came up with the similar structure. I asked because I don't like this overhead. But I understand there are some Swift's nuances which leads to difficulties in such kind of API.
